api/handlers/datasources.py

```diff
@@ -47,6 +51,38 @@ def get_datasources(
     return result
 
 
+@router.get(
+    "/api/datasources/{account}/{container}/image", response_class=StreamingResponse
+)
+async def get_datasource_image(
+    account: str,
+    container: str,
+    datasources_collection: Collection[DataSource] = Depends(
+        database.database.datasources_collection
+    ),
+):
+    # Create the imageURL with sasToken
+    datasource = datasources_collection.find_one(
+        {
+            "account": account,
+            "container": container,
+        }
+    )
+    if not datasource:
+        raise HTTPException(status_code=404, detail="Datasource not found")
+
+    imageURL = add_URL_sasToken(account, container, datasource["sasToken"], "", uiImage.IMAGE)
+    if not imageURL.get_secret_value():
+        return StreamingResponse((b'' for _ in range(1))) # return empty image
```

Comment:
Strangest way that I have even seen to make an empty image but if it works it works.
But shouldnt we send a defautl image in case it is empty ?